app/herdbook.py

```diff
@@ -523,20 +529,26 @@ def issue_certificate(i_number):
     if certificate_exists:
         return jsonify({"response": "Certificate already exists"}), 400
 
-    ## TODO: Make sure this is unique
-    cert_number = str(random.randint(50001, 99999))
+    ##cert_number = str(random.randint(50001, 99999))
+    cert_number = "99999"
     ind.update(**request.form, certificate=cert_number)
     da.update_individual(ind, session.get("user_id", None))
 
     data = get_certificate_data(ind, user_id)
     pdf_bytes = get_certificate(data)
     ind_number = ind["number"]
-    uploaded = upload_certificate(
-        pdf_bytes=pdf_bytes.getvalue(), ind_number=ind_number
-    )
+
+    try:
+        signed_data = sign_data(pdf_bytes)
+        uploaded = upload_certificate(
+            pdf_bytes=signed_data.getvalue(), ind_number=ind_number
+        )
+    except Exception as ex:  # pylint: disable=broad-except
+        print(ex)
+        return jsonify({"response": "Error processing your request"}), 400
 
     if uploaded:
```

Comment:
I'm not overly fond of this style because one needs to think about the flow to understand why this doesn't blow up with `uploaded` undefined. I much prefer either having the return within the try or making sure to define `uploaded` to `None` above the try.

Reply:
This one is for capturing mostly S3 related failures, which are a bit tricky to propagate otherwise. I am happy to work on a better solution in the next iteration :)

Reply:
I think I may have been unclear, I'd be happy if there would be an `uploaded = False` above the code.

As far as I can see, the code above works (won't blow up on `uploaded` undefined), but it isn't obvious.

Reply:
Aha! Alright now I follow you.

Reply:
Should be fixed now